src/model.ts

```diff
@@ -11,14 +11,26 @@ export class Model {
    * @param inputs - one or more names of the model's input nodes (applicable only for TensorFlow models)
    * @param outputs - one or more names of the model's output nodes (applicable only for TensorFlow models)
    * @param blob - the Protobuf-serialized model
+   * @param batchsize - when provided with an batchsize that is greater than 0, the engine will batch incoming requests from multiple clients that use the model with input tensors of the same shape.
+   * @param minbatchsize -  when provided with an minbatchsize that is greater than 0, the engine will postpone calls to AI.MODELRUN until the batch's size had reached minbatchsize
    */
-  constructor(backend: Backend, device: string, inputs: string[], outputs: string[], blob: Buffer | undefined) {
+  constructor(
+    backend: Backend,
+    device: string,
+    inputs: string[],
+    outputs: string[],
+    blob: Buffer | undefined,
+    batchsize?: number,
+    minbatchsize?: number,
+  ) {
     this._backend = backend;
     this._device = device;
     this._inputs = inputs;
     this._outputs = outputs;
     this._blob = blob;
     this._tag = undefined;
+    this._batchsize = batchsize || 0;
```

Comment:
maybe verify batchsize and minbatchsize are >=0?

Reply:
adding a guard to it after initial check for undefined or assignement to zero.
```
this._batchsize = batchsize || 0;
    if (this._batchsize < 0) {
      this._batchsize = 0;
    }
```